### PR TITLE
design-system(mobile): SureDialog primitive + migrate all AlertDialog usages

### LINKED ISSUE #2454: design-system(mobile): SureDialog primitive (tokenized modal replacing Material AlertDialog)
Part of the mobile design-system sequence (#2235). After the surface, money, icon, and form-control primitives landed, dialogs are the last large cluster of un-tokenized Material chrome in the app.

Every confirmation and info dialog today uses Material `AlertDialog`: the default M3 chrome (corner radius, elevation, title/body typography) with raw `TextButton`/`ElevatedButton`/`FilledButton` actions. This ignores the Sure palette and `SureButton`, so confirmations read off-brand and — in dark mode especially — inconsistent with the already-redesigned surfaces. It also blocks the planned off-Material end phase (dropping `MaterialApp` and Material-styled widgets), which cannot proceed while Material dialogs remain.

**Scope:** add a `SureDialog` primitive — a tokenized `container` surface (hairline `borderSecondary` border, the subtle DS elevation shadow, `radiusLg`) with a semibold `textPrimary` title, an optional body (`message` text or arbitrary `content`), and a trailing actions row that wraps when it doesn't fit — resolving from the active `SureColors` palette so it is brightness-aware and stays in lockstep with `sure.tokens.json`. Add a `SureDialog.confirm` helper for the dominant confirm/cancel pattern, with `destructive` and `confirmEnabled` flags so per-site behavior is preserved. Migrate every Material `AlertDialog` in the app onto it (settings, transactions list, chat list, main navigation, login, chat conversation, calendar, log viewer). Tests + before/after screenshots.

This removes the last Material `AlertDialog`s from the app.

### PR DESCRIPTION
**Pull request by JSONbored**
## What

Adds a **`SureDialog`** design-system primitive — a tokenized replacement for Material's `AlertDialog` — and migrates **every** `AlertDialog` in the app onto it: 16 sites across 8 screens. After this change there are no Material `AlertDialog`s left in the mobile app.

Closes #2454. Part of #2235.

## Why

Dialogs were the last large cluster of un-tokenized Material chrome: default M3 corner radius/elevation/typography with raw `TextButton`/`ElevatedButton`/`FilledButton` actions, ignoring the Sure palette and `SureButton`. This read off-brand and, in dark mode, inconsistent with the already-redesigned surfaces. It is also the prerequisite for the off-Material end phase (dropping `MaterialApp` and Material-styled widgets), which can't proceed while Material dialogs remain.

## How

- `lib/widgets/sure_dialog.dart` mirrors the `SureCard`/`SureButton` pattern: a rounded `container` surface with a hairline `borderSecondary` border and the DS elevation shadow, a semibold `textPrimary` title, an optional body (`message` text or arbitrary `content`), and a trailing actions row that wraps to a vertical stack when it doesn't fit. Colors resolve from the active `SureColors` palette, so it is brightness-aware and stays in lockstep with `sure.tokens.json`.
- A `SureDialog.confirm(...)` helper collapses the dominant confirm/cancel pattern to a one-liner returning `Future<bool?>`, with `destructive` and `confirmEnabled` flags that preserve each call site's exact behavior.
- Migrated all sites — settings (5 confirms + 1 form), transactions list (3), chat list (2), main navigation (1), login (1, custom), chat conversation (1, rename input → `SureTextField`), calendar (1, info), log viewer (1) — preserving localization strings, return values, and destructive/disabled states. This is a chrome swap, not a behavior change.

## Screenshots

Destructive confirm ("Delete account"), captured from the running app on the iOS Simulator (iPhone 17 Pro):

<table>
  <thead><tr><th>Mode</th><th>Before — Material <code>AlertDialog</code></th><th>After — <code>SureDialog</code></th></tr></thead>
  <tbody>
    <tr>
      <td><strong>Light</strong></td>
      <td><a href="https://raw.githubusercontent.com/JSONbored/sure-upstream/codex/pr-evidence/docs/pr-assets/mobile-sure-dialog/before-dialog-light.png">Before – light</a><br><em>Flat Material chrome, text-only actions</em></td>
      <td><a href="https://raw.githubusercontent.com/JSONbored/sure-upstream/codex/pr-evidence/docs/pr-assets/mobile-sure-dialog/after-dialog-light.png">After – light</a><br><em>Tokenized surface, ghost Cancel + destructive SureButton</em></td>
    </tr>
    <tr>
      <td><strong>Dark</strong></td>
      <td><a href="https://raw.githubusercontent.com/JSONbored/sure-upstream/codex/pr-evidence/docs/pr-assets/mobile-sure-dialog/before-dialog-dark.png">Before – dark</a><br><em>Material surface drifts from the redesigned dark theme</em></td>
      <td><a href="https://raw.githubusercontent.com/JSONbored/sure-upstream/codex/pr-evidence/docs/pr-assets/mobile-sure-dialog/after-dialog-dark.png">After – dark</a><br><em>Dark <code>container</code> token, hairline border, dark-mode destructive red</em></td>
    </tr>
  </tbody>
</table>

## Testing

- `mobile/test/widgets/sure_dialog_test.dart` (8 new cases): token chrome resolves the right palette in light **and** dark (container/border/radius/shadow), the `message`/`content` assertion, `confirm` resolves true on confirm and false on cancel, `destructive` renders the destructive variant, `confirmEnabled: false` disables the confirm action, and the custom-content path renders content + trailing actions.
- `flutter analyze` — clean (the only 3 infos are pre-existing in `intro_screen_web.dart`, untouched here).
- `flutter test` — green (174, incl. the 8 above).
- iOS debug build/run verified on the simulator (screenshots above). No dependency changes.

## Summary by CodeRabbit

* **New Features**
  * Introduced custom-styled `SureDialog` and `SureButton`/`SureTextField` components, and migrated confirmation dialogs across the app for consistent look and behavior (including destructive and disabled-confirm states).

* **Bug Fixes**
  * Ensured confirm/cancel flows return the correct results across dialog dismissals (null on dismiss, `true/false` on confirm/cancel), preserving existing action logic.

* **Tests**
  * Added comprehensive widget tests for `SureDialog`, including light/dark theme styling, validation rules, confirm/cancel outcomes, destructive variant styling, and scroll behavior for oversized content.

### CODE REVIEW

File: mobile/lib/widgets/sure_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+              title,
+              style: TextStyle(
+                fontSize: 18,
+                fontWeight: FontWeight.w600,
+                color: palette.textPrimary,
+              ),
+            ),
+            if (body != null) ...[
+              const SizedBox(height: 12),
+              body,
```
[the remark this answers is not in the document]
Comment (JSONbored) [reply mobile/lib/widgets/sure_dialog.dart:136]: Good catch — fixed in 3e03d8b02. The body is now a `Flexible` (loose) child of the Column, so short content sizes naturally while tall content is capped to the available dialog height and scrolls within it (matching the AlertDialog content area it replaced). Caller-supplied `content` keeps its own scrollable (calendar list, headers form); a plain `message` is wrapped in a `SingleChildScrollView`. Added a regression test that asserts oversized content stays bounded and scrollable with no overflow.

**Review by jjmata — COMMENTED** [review]
This is a well-executed primitive. The API is clean — `SureDialog` for arbitrary content, `SureDialog.confirm` for the common two-button case — and the migration across all the `AlertDialog` call sites is thorough.

A few small observations:

**`message` body font size is a raw `14`** — In `SureDialog.build`, the message `TextStyle` uses `fontSize: 14` directly. Given `SureTypography.sm = 14` now exists (from PR #2438), this could use `SureTypography.sm` for consistency, but that PR may not be merged yet — coordinate accordingly.

**`_showUpdateDialog` `confirmEnabled` mapping** — The old `TextButton` had `onPressed: storeUrl == null ? null : () => Navigator.pop(ctx, true)`. The new `SureDialog.confirm` call uses `confirmEnabled: storeUrl != null`. These are equivalent; good.

**Tests are comprehensive** — the brightness-aware chrome test, `confirm` resolve/cancel/destructive/disabled paths, oversized-content overflow test, and the `assert` for mutually exclusive `message`/`content` are all covered. The only thing missing is a test that the barrier tap (outside the dialog) resolves to `null` rather than `false`, but that's testing Flutter framework behaviour more than your widget, so it's fine to omit.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by jjmata — CHANGES_REQUESTED** [review]
Merge conflicts here too, @JSONbored

**Comment by sure-design** [comment]
Periodic PR-health check: this PR currently shows `mergeable_state: dirty` (merge conflicts with `main`), consistent with @jjmata's June 30 review ("Merge conflicts here too, @JSONbored"). That review is still outstanding with no follow-up commit or reply since. Flagging so this doesn't get lost — otherwise the SureDialog work here looks solid (tokenized chrome, thorough AlertDialog migration, and good test coverage for confirm/cancel/destructive/disabled paths).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sure-design** [comment]
Heads up: @jjmata requested changes on 2026-06-30 citing merge conflicts on this branch, and there's been no follow-up commit or reply since. Could you rebase/resolve so this can get re-reviewed?

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sure-design** [comment]
**Recommendation: request-changes**

The `SureDialog` primitive itself looks good — tokenized chrome mirroring `SureCard`/`SureButton`, brightness-aware, thorough migration of all 16 `AlertDialog` sites, and solid widget-test coverage (light/dark chrome, confirm/cancel/destructive/disabled paths, overflow scrolling). @jjmata's substantive review on 2026-06-23 was positive with only minor notes.

However, @jjmata's 2026-06-30 review requested changes citing merge conflicts (`mergeable_state` is currently `dirty`), and there's been no follow-up commit or reply from the author since then (last commit is 2026-06-22/23, over 3 weeks old). Please rebase onto `main` and resolve the conflicts so this can get re-reviewed and merged — the design-system work itself appears ready.

_Automated review via Claude Code PR-triage routine._

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sure-admin** [comment]
Hi @JSONbored, this PR currently has merge conflicts with the base branch. Please rebase or merge the latest base branch and resolve the conflicts when you get a chance.

As part of the PR backlog cleanup, maintainers plan to close PRs that still need author action in one week. If you want to keep working on this or need more time, just comment here and we will leave it open. Thanks!

**Comment by jjmata** [comment]
Thanks for the contribution, and apologies this sat so long. I'm doing a long-overdue queue cleanup and closing PRs that have been inactive for 60+ days and/or conflict with main. That's housekeeping, not a judgment on the work - if you want to pick it back up, rebase and reopen (or a fresh PR) is very welcome anytime.
